src/test/ui/simd/issue-105439.rs

```diff
@@ -0,0 +1,14 @@
+// This is used to ICE with MIR inlining enabled due to an invalid bitcast.
+// run-pass
+// compile-flags: -O -Zmir-opt-level=3
```

Comment:
Perhaps a smaller test would capture the essence of the issue without requiring MIR optimizations:

```rust
#![feature(repr_simd)]
#![feature(platform_intrinsics)]

#[allow(non_camel_case_types)]
#[derive(Clone, Copy)]
#[repr(simd)]
pub struct i32x4([i32; 4]);

pub fn f(a: i32x4) -> [i32; 4] {
    let b = a;
    b.0
}
```

```console
$ rustc --crate-type=lib a.rs -Zverify-llvm-ir
Invalid bitcast
  %4 = bitcast <4 x i32> %3 to [4 x i32]
LLVM ERROR: Broken module found, compilation aborted!
```

Reply:
Sure, updated test.